### PR TITLE
chore!: convert to typescript and vitest

### PR DESCRIPTION
**Pull request by westeezy**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,22 +2,29 @@
   "name": "@krakenjs/webpack-mem-compile",
   "version": "2.0.2",
   "description": "Webpack compile to memory",
-  "main": "index.js",
+  "main": "dist/esm/index.js",
```
Comment (westeezy) [inline package.json:5]: we will need to test esm support in csdk

File: .npmrc [hunk 2]
```diff
@@ -1,2 +1 @@
 registry=https://registry.npmjs.org/
-package-lock=false
```
Comment (dtjones404) [inline .npmrc:2]: Should we be committing the new lockfile?

Comment (wsbrunson) [reply .npmrc:2]: this repo is fine because it is used server-side but good catch

**Review by dtjones404 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (westeezy) [reply package.json]: dropping esm for this iteration as the side car only accepts cjs

**Pull request merged**